app/utils/store.py

```diff
@@ -26,6 +30,23 @@ def __init__(self, bucket=None):
     def init_app(self, app):
         self.bucket = app.config["DOCUMENTS_BUCKET"]
 
+    def check_for_blocked_document(self, service_id, document_id):
+        """Raises an exception if access to the document has been blocked after creation
+
+        This should be checked before any document access. This might be used to quickly prevent anyone from accessing
+        a file that a service has sent out accidentally.
+
+        Note that the `blocked` tag key MUST be in lowercase.
+        """
+        tags = {
+            item["Key"]: item["Value"]
+            for item in self.s3.get_object_tagging(
+                Bucket=self.bucket, Key=self.get_document_key(service_id, document_id)
+            )["TagSet"]
```

Comment:
into the ether: it's kinda annoying that we use both tagging and metadata here. I guess tagging makes sense as it fits a tag (you could use this to group files into blocked and not blocked, for example), but metadata is nice because it keeps tags to just the retention period (read and acted on by AWS) vs the metadata as all the business logic we control in code. thoughts?

Reply:
Metadata would be nice, but: "You can set object metadata in Amazon S3 at the time you upload the object. Object metadata is a set of name-value pairs. After you upload the object, you cannot modify object metadata. The only way to modify object metadata is to make a copy of the object and set the metadata."

This means that in order to block a document we'd need to upload a new version, which means we would need the encryption key, which means we can't do it via the console. It makes the whole process more complicated basically.

Reply:
makes sense